File: 406-queueReconstructionHeight/solution.java
import java.util.ArrayList;

class Solution {
    public int[][] reconstructQueue(int[][] people) {

        // 26%
        Arrays.sort(people, (a, b) -> 
                    (a[0] == b[0] ? a[1]-b[1]: b[0]-a[0]));
        ArrayList<int[]> list = new ArrayList<>();
        
        for (int[] p : people){
            list.add(p[1], p);
        }
        
        return list.toArray(new int[people.length][2]);
    }
}

File: 406-queueReconstructionHeight/solution.py
class Solution:
    def reconstructQueue(self, people):
        """
        :type people: List[List[int]]
        :rtype: List[List[int]]
        """
        # sort by height descending & numPeople ascending O(N*log(N)), 82%
        length = len(people)

        if length<2:
            return people

        # people=sorted(people ,key=lambda l:(l[0], -l[1]), reverse=True)
        people=sorted(people ,key=lambda l:(-l[0], l[1]))

        result = []
        for person in people:
            # print(result)
            result.insert(person[1], person)

        return result